### PR TITLE
Key Vault breaking changes

### PR DESCRIPTION
**Pull request by maddieclayton**
## Description
breaking changes described here: https://github.com/Azure/azure-powershell/issues/5496

Also handles this issue: https://github.com/Azure/azure-powershell/issues/4393, which is a breaking change to the client.

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
@cormacpayne Changed the some of the sets to updates - kept Set-AzureRmKeyVaultAccessPolicy the same, as there is no add/new.

File: src/ResourceManager/KeyVault/AzureRM.KeyVault.psd1 [hunk 1]
```diff
@@ -100,8 +100,8 @@ CmdletsToExport = 'Add-AzureKeyVaultCertificate',
                'Undo-AzureKeyVaultKeyRemoval', 'Undo-AzureKeyVaultSecretRemoval', 
                'Add-AzureKeyVaultKey', 'Remove-AzureKeyVaultKey', 
                'Remove-AzureKeyVaultSecret', 'Restore-AzureKeyVaultKey', 
-               'Set-AzureKeyVaultKeyAttribute', 'Set-AzureKeyVaultSecret', 
-               'Set-AzureKeyVaultSecretAttribute', 
+               'Update-AzureKeyVaultKey', 'Set-AzureKeyVaultSecret',
+			   'Update-AzureKeyVaultSecret',
```
Comment (cormacpayne) [inline src/ResourceManager/KeyVault/AzureRM.KeyVault.psd1]: @maddieclayton nit: I would run `Update-ModuleManifest` on this file to fix the formatting

File: src/ResourceManager/KeyVault/Commands.KeyVault/help/Update-AzureKeyVaultKey.md [hunk 2]
```diff
@@ -14,16 +14,16 @@ Updates the attributes of a key in a key vault.
...
 ```
-Set-AzureKeyVaultKeyAttribute [-InputObject] <PSKeyVaultKeyIdentityItem> [[-Version] <String>]
- [-Enable <Boolean>] [-Expires <DateTime>] [-NotBefore <DateTime>] [-KeyOps <String[]>] [-Tag <Hashtable>]
- [-PassThru] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+Update-AzureKeyVaultKey [-InputObject] <PSKeyVaultKeyIdentityItem> [[-Version] <String>] [-Enable <Boolean>]
+ [-Expires <DateTime>] [-NotBefore <DateTime>] [-KeyOps <String[]>] [-Tag <Hashtable>] [-PassThru]
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
 ## DESCRIPTION
```
Comment (cormacpayne) [inline src/ResourceManager/KeyVault/Commands.KeyVault/help/Update-AzureKeyVaultKey.md]: @maddieclayton cmdlet name wasn't changed in the description below

File: src/ResourceManager/KeyVault/Commands.KeyVault/help/Undo-AzureRmKeyVaultRemoval.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ Undo-AzureRmKeyVaultRemoval [-VaultName] <String> [-ResourceGroupName] <String>
 
 ### InputObject
 ```
-Undo-AzureRmKeyVaultRemoval [-InputObject] <PSDeletedKeyVault> [-ResourceGroupName] <String> [-Tag <Hashtable>]
```
Comment (cormacpayne) [inline src/ResourceManager/KeyVault/Commands.KeyVault/help/Undo-AzureRmKeyVaultRemoval.md]: @maddieclayton `ResourceGroupName` was removed, but still shows up in the example and in the list of parameters below

Comment (cormacpayne) [reply src/ResourceManager/KeyVault/Commands.KeyVault/help/Undo-AzureRmKeyVaultRemoval.md]: @maddieclayton oops ignore this

File: src/ResourceManager/KeyVault/Commands.KeyVault/help/Update-AzureKeyVaultKey.md (hunk 2 above)
Comment (maddieclayton) [reply src/ResourceManager/KeyVault/Commands.KeyVault/help/Update-AzureKeyVaultKey.md]: Done

File: src/ResourceManager/KeyVault/AzureRM.KeyVault.psd1 (hunk 1 above)
Comment (maddieclayton) [reply src/ResourceManager/KeyVault/AzureRM.KeyVault.psd1]: Done

**Comment by maddieclayton** [comment]
on demand here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/694/ (and 695)

**Pull request merged**